src/main/java/uk/gov/pay/directdebit/webhook/gocardless/api/GoCardlessWebhookParser.java

```diff
@@ -32,16 +32,63 @@ public GoCardlessWebhookParser(ObjectMapper objectMapper) {
             JsonNode webhookJson = objectMapper.readTree(webhookPayload);
             JsonNode eventsPayload = webhookJson.get("events");
             for (JsonNode eventNode : eventsPayload) {
-                String resourceType = eventNode.get("resource_type").asText();
+                JsonNode detailsNode = eventNode.get("details");
+                JsonNode linksNode = eventNode.get("links");
+                String detailsCause =  
+                        detailsNode.has("cause") ? detailsNode.get("cause").asText() : null;
+                String detailsDescription =
+                        detailsNode.has("description") ? detailsNode.get("description").asText() : null;
+                String detailsOrigin =
+                        detailsNode.has("origin") ? detailsNode.get("origin").asText() :null;
+                String detailsReasonCode =
+                        detailsNode.has("reason_code") ? detailsNode.get("reason_code").asText() :null;
+                String detailsScheme =
+                        detailsNode.has("scheme") ? detailsNode.get("scheme").asText() :null;
+                String linksMandate 
+                        = detailsNode.has("mandate") ? linksNode.get("mandate").asText() : null;
+                String linksNewCustomerBankAccount 
+                        = detailsNode.has("new_customer_bank_account") 
+                        ? linksNode.get("new_customer_bank_account").asText() : null;
+                String linksNewMandate 
+                        = detailsNode.has("new_mandate") ? linksNode.get("new_mandate").asText() : null;
+                String linksOrganisation
+                        = detailsNode.has("organisation") ? linksNode.get("organisation").asText() : null;
+                String linksParentEvent 
+                        = detailsNode.has("parent_event") ? linksNode.get("parent_event").asText() : null;
+                String linksPayment 
+                        = detailsNode.has("payment") ? linksNode.get("payment").asText() : null;
+                String linksPayout 
+                        = detailsNode.has("payout") ? linksNode.get("payout").asText() : null;
+                String linksPreviousCustomerBankAccount = detailsNode.has("previous_customer_bank_account") 
+                        ? linksNode.get("previous_customer_bank_account").asText() : null;
+                String linksRefund 
+                        = detailsNode.has("refund") ? linksNode.get("refund").asText() : null;
+                String linksSubscription = detailsNode.has("subscription") 
+                        ? linksNode.get("subscription").asText() : null;
+                        String resourceType = eventNode.get("resource_type").asText();
```

Comment:
Since there are three overloaded constructors for `GoCardlessEvent` each taking about 20 parameters and many being optional perhaps now is a good time to introduce a builder for `GoCardlessEvent`?